lib/Twig/Cache/Filesystem.php

```diff
@@ -58,8 +58,11 @@ public function write($key, $content)
     {
         $dir = dirname($key);
         if (!is_dir($dir)) {
-            if (false === @mkdir($dir, 0777, true) && !is_dir($dir)) {
-                throw new RuntimeException(sprintf('Unable to create the cache directory (%s).', $dir));
+            if (false === @mkdir($dir, 0777, true)) {
+                clearstatcache(false, $dir);
+                if (!is_dir($dir)) {
+                    throw new RuntimeException(sprintf('Unable to create the cache directory (%s).', $dir));
+                }
```

Comment:
If this is done to deal with a race condition where the directory is made by another process  than the directory might not be writable at this point, which is not checked as is done on line 67.

The call to [clearstatcache](http://php.net/manual/en/function.clearstatcache.php) will not do what you expected, as the second param (`$dir`) is only used if the first param is `true`.